pkg/controller/gittrack/gittrack_controller.go

```diff
@@ -543,22 +548,38 @@ func checkOwner(owner farosv1alpha1.GitTrackInterface, child farosv1alpha1.GitTr
 }
 
 // ignoreObject checks whether the unstructured object should be ignored
-func (r *ReconcileGitTrack) ignoreObject(u *unstructured.Unstructured) (bool, string, error) {
+func (r *ReconcileGitTrack) ignoreObject(u *unstructured.Unstructured, owner farosv1alpha1.GitTrackInterface) (bool, string, error) {
 	gvr, namespaced, err := utils.GetAPIResource(r.restMapper, u.GetObjectKind().GroupVersionKind())
 	if err != nil {
 		return false, "", err
 	}
 
 	// Ignore namespaced objects not in the namespace managed by the controller
-	if namespaced && farosflags.Namespace != "" && farosflags.Namespace != u.GetNamespace() {
-		r.log.V(1).Info("Object not in namespace", "object namespace", u.GetNamespace(), "managed namespace", farosflags.Namespace)
+	if namespaced && r.namespace != "" && r.namespace != u.GetNamespace() {
+		r.log.V(1).Info("Object not in namespace", "object namespace", u.GetNamespace(), "managed namespace", r.namespace)
 		return true, fmt.Sprintf("namespace `%s` is not managed by this Faros", u.GetNamespace()), nil
 	}
 	// Ignore GVKs in the ignoredGVKs set
 	if _, ok := r.ignoredGVRs[gvr]; ok {
 		r.log.V(1).Info("Object group version ignored globally", "group version resource", gvr.String())
 		return true, fmt.Sprintf("resource `%s.%s/%s` ignored globally by flag", gvr.Resource, gvr.Group, gvr.Version), nil
 	}
+
+	ownerNamespace := owner.GetNamespace()
+	_, ownerIsGittrack := owner.(*farosv1alpha1.GitTrack)
+	if namespaced {
+		// prevent a gittrack in a namespace from handling objects which are in a different namespace
+		if ownerIsGittrack && ownerNamespace != u.GetNamespace() {
+			return true, fmt.Sprintf("namespace `%s` is not managed by this GitTrack", u.GetNamespace()), nil
+		} else if !ownerIsGittrack && r.clusterGitTrackMode == farosflags.CGTMExcludeNamespaced {
+			return true, "namespaced resources cannot be managed by ClusterGitTrack", nil
+		}
+	}
+
+	if !ownerIsGittrack && r.clusterGitTrackMode == farosflags.CGTMDisabled {
```

Comment:
Right now, I've just implemented disabling CGT as ignoring any component within one, but I intend to change this, so that we don't even look at the objects within when disabled